### PR TITLE
PLAT-7015: Django Modules as Python Packages

### PR DESCRIPTION
**Pull request by math-a3k**
- Update modules.utils
- Update modules.admin
- Update modules.urls

Implementation Details
======================

In order to support Django Modules as Python Packages, modules app's
utils, admin and urls discovery must be generalized to any subdirectory.

This is not a breaking change, as current Django Modules should work.

### CODE REVIEW

File: {{cookiecutter.project_slug}}/modules/utils.py [hunk 1]
```diff
@@ -6,6 +6,12 @@
 GLOBAL_OPTIONS_FILE_PATH = f"{Path.cwd()}/modules/options.json"
 
 
+def posixpath_to_modulepath(posixpath):
+    parent_mondule_path = posixpath.parent.as_posix().split("/")
+    module_name = [posixpath.stem, ]
+    return ".".join(parent_mondule_path + module_name)
```
Comment (danielsousaio) [inline {{cookiecutter.project_slug}}/modules/utils.py:12]:
```suggestion
    parent_module_path = posixpath.parent.as_posix().split("/")
    module_name = [posixpath.stem, ]
    return ".".join(parent_module_path + module_name) 
```

Comment (danielsousaio) [inline {{cookiecutter.project_slug}}/modules/utils.py:12]:
```suggestion
    return f"{parent_module_path}.{module_name}"
```
Perhaps an f-string here is more readable?

File: {{cookiecutter.project_slug}}/Pipfile [hunk 2]
```diff
@@ -11,7 +11,7 @@ python_version = "3.8"
 [packages]
 cryptography = "~=37.0.2"
 django = "~=2.2.26"
-"boto3" = "~=1.24.13"
+"boto3" = "~=1.24.22"
```
Comment (danielsousaio) [inline {{cookiecutter.project_slug}}/Pipfile:14]: This package is not related to this PR, perhaps we should leave it untouched?

File: {{cookiecutter.project_slug}}/modules/utils.py (hunk 1 above)
Comment (math-a3k) [reply {{cookiecutter.project_slug}}/modules/utils.py]: parent_module_path is a list, to return with a f-string it would have to be:
```
parent_module_path = ".".join(posixpath.parent.as_posix().split("/"))
module_name = posixpath.stem
return f"{parent_module_path}.{module_name}"
```
perhaps it's a bit more readable, something like:
```
def posixpath_to_modulepath(posixpath):
    module_parent_path = ".".join(posixpath.parent.as_posix().split("/"))
    return f"{module_parent_path}.{posixpath.stem}"
```
?

File: {{cookiecutter.project_slug}}/Pipfile (hunk 2 above)
Comment (math-a3k) [reply {{cookiecutter.project_slug}}/Pipfile:14]: Yes, it would be better handle dependencies update in other PR

Comment (math-a3k) [reply {{cookiecutter.project_slug}}/Pipfile]: This was because of a rebase, a dependabot commit ended up the branch - now the rebase should be proper

File: {{cookiecutter.project_slug}}/modules/utils.py (hunk 1 above)
Comment (danielsousaio) [reply {{cookiecutter.project_slug}}/modules/utils.py]: Last one looks very easy to read and understand. If that is working I would suggest that one

File: {{cookiecutter.project_slug}}/Pipfile (hunk 2 above)
Comment (danielsousaio) [reply {{cookiecutter.project_slug}}/Pipfile]: Okay, great. Thanks!

**Review by danielsousaio — APPROVED** [review]
